bugbug/db.py

```diff
@@ -54,6 +54,23 @@ def is_old_schema(path):
     return DATABASES[path]["version"] > prev_version
 
 
+# used almost same code as is_old_schema function with small change at return statement.
+# added function docstring.
+def is_db_version_changed(path):
```

Comment:
Instead of introducing a new function, I think we can just use `is_old_schema` in `last_modified`

Reply:
But, it only checks if db is old or not.
If db is new then only it return true.
But here we need to check if db is changed to old or new version.

Reply:
It should basically never happen as we always increase version numbers and never decrease them, but to account for the remote possibility we can simply change the `is_old_schema` check to use `!=` instead of `>`.

Reply:
If we replace > with != It will not the same what function name is_old_schema implies(meaning).

Reply:
We can change its name to something like `is_different_schema`

Reply:
Ok

Reply:
But look at at first 3 lines of download function.
If we use is_different_schema with != in there, 
then there is no download for both old and new db versions.

Reply:
If the schema version is newer than the schema of the database that is available at the URL, we don't want to download (this is the same behavior as before).
If the schema version is older than the schema of the database that is available at the URL, we also don't want to download (this is not the same behavior as before, but we do want this change of behavior).

Reply:
I didn't understand the second point

Reply:
We have two schema versions, the version that is the local one (the one bugbug is currently using) and the version that is the remote one (the one bugbug used to generate the DB and upload it at the URL).

When we change the version, there will be a mismatch between the local and the remote schema versions and bugbug will have to regenerate the DB from scratch (since the schema is different, bugbug won't be able to use the previous DB that was uploaded at the URL).

Reply:
Ok.
What I understand is :
1. The version with which bugbug previously run is stored in url location and is the old version.
2. When the bugbug  run again locally with same old version then the schema will get downloaded from that previous url.(according to download function.)
3. If used even older versions ,then no download.(download function return false.)
4. Now again when bugbug is run locally with new version, download function is returning false.

So, please clarify what to do if new version is used when bugbug is run again locally ?
Just return last modified as none if new version is used ?

Reply:
Is this is what actually happening ?

Reply:
Let me make an example with numbers so it will hopefully be clearer.

There is a run which generates the DB with the current schema (let's say version 7).

Now, there can be three cases for the following run:
- The schema hasn't changed, so the version is still 7. We can reuse the previous DB, so we download it.
- The schema has been updated, so the version is 8. We can't reuse the previous DB, so we don't download it.
- The schema has been reverted to a previous state, so the version is 6. As above, we can't reuse the previous DB, so we don't download it.

Reply:
Ok, now I understand clearly.
Thank you.

Reply:
ok then
Just return last modified as None if new version ? To fix #2459

Reply:
Return None if different (so either new or old)

Reply:
Ok